lib/bom.js

```diff
@@ -510,7 +510,7 @@ declare class WorkerGlobalScope extends EventTarget {
 }
 
 declare class DedicatedWorkerGlobalScope extends WorkerGlobalScope {
```

Comment:
you need to override `self` here to ensure that `self` in the `DedicatedWorkerGlobalScope` refers to `DedicatedWorkerGlobalScope` with `self: DedicatedWorkerGlobalScope`

Reply:
Otherwise if someone writes self.postMessage() in a WebWorker, it is still going to resolve to the definition for the WorkerGlobalScope.